src/main/java/de/tum/in/www1/artemis/repository/GradeStepRepository.java

```diff
@@ -0,0 +1,21 @@
+package de.tum.in.www1.artemis.repository;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import de.tum.in.www1.artemis.domain.GradeStep;
+
+/**
+ * Spring Data JPA Repository for the GradeStep entity
+ */
+@Repository
+public interface GradeStepRepository extends JpaRepository<GradeStep, Long> {
+
+    List<GradeStep> findByGradingScale_Id(Long gradingScaleId);
+
+    Optional<GradeStep> findByIdAndGradingScale_Id(Long gradeStepId, Long gradingScaleId);
```

Comment:
Does this want an `OrElseThrow` method?

Reply:
I could add the change, those methods only get used in the grade step resources when querying for a specific grade step by id. The check whether this query returns a grade step or is empty occurs in the resource the way it is implemented now